lib/firebase_constants.dart

```diff
@@ -27,3 +27,5 @@ String get themesDoc => _constants['themesDoc'];
 String get dataDoc => _constants['dataDoc'];
 String get peopleCollection => _constants['peopleCollection'];
 String get messagesCollection => _constants['messagesCollection'];
+List<String> get allowedEmailDomains =>
```

Comment:
I don't really follow your intention here - this configuration needs to be done on the server not the client - let me know if that doesn't make sense and you need a hand with how we setup this kind of thing

Reply:
The actual access control will be done on the server. This is purely to show an error message in the client that says along the lines of "the email under this domain is not permitted". So when the user signs in with say @gmail.com - I can catch the email to see if it ends with the domains in the config and show the notification in the client.

Is this the way to do this or is there any other way to whitelist allowed email domains for authentication? @lukechurch

Reply:
In reviewing the other PR I ended up looking into this as well. I didn't know this until now, but it turns out that the way Firebase and Google OAuth integrate, any user can "create an account" by logging in by default, and you need to have multiple checks to support the kind of restricted logging in we're trying to do.

[This stackoverflow answer](https://stackoverflow.com/a/44347561) summarises what needs to be done: we need the firebase rules to specify read/writes on the database, but we also need to handle users that go through the login process but who are not allowed to read/write (ie. the domain filtering), but who have been logged in. In Coda we show that there are no datasets available by having an empty list, or show the missing permissions error when trying to access a project that the user doesn't have permissions to access, while Nook doesn't really handle this other than throwing errors in the console.

So from what I can tell, checking that the email of the logged in user is part of an allowed domain in the client like this seems to be the recommended solution, even if it's kinda ugly.

@elayabharath May I suggest adding a comment here that `allowedEmailDomains` should match the list of allowed domains in the firebase rules file?

Reply:
Thanks for the summary. Added the comment in the place where we read the domains since json doesn't have comments.